src/bin/dcr.rs

```diff
@@ -0,0 +1,56 @@
+#[macro_use]
+extern crate derive_new;
+#[macro_use]
+extern crate serde_derive;
+extern crate ctrlc;
+
+use crate::docker_cluster_runner::fs_configuration::FSConfiguration;
+use crate::docker_cluster_runner::TestClusterConfiguration;
+use clap::{App, Arg};
+use std::error::Error;
+use std::fs;
+use std::sync::{Arc, Mutex};
+
+mod docker_cluster_runner;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let config_dir = "cluster_test";
+    let fs_configuration = FSConfiguration::new(
+        ".".to_string(),
+        config_dir.to_string(),
+        config_dir.to_string(),
+        "/tmp".to_string(),
+    );
+    let configuration = get_configuration()?;
+    configuration.save_cluster_configuration(config_dir)?;
+    let compose = configuration.create_docker_compose(fs_configuration, "bobnet".to_string())?;
+    let arc = Arc::new(Mutex::new(Box::new(compose)));
+    let ctrlc_arc = arc.clone();
+    ctrlc::set_handler(move || {
+        let compose = ctrlc_arc.lock().unwrap();
```

Comment:
Is it possible thread lock here?

Reply:
Actually no, I'll introduce new var so it will be clear